src/no-test-import.ts

```diff
@@ -0,0 +1,54 @@
+// no-test-import.ts
+
+import type { Rule } from 'eslint';
+
+/*
+ * Copyright (c) 2021-2024 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+/**
+ *  this rule should be turned on with overrides for test files (e.g. *.test.ts, *.spec.ts) only
+ */
+
+export interface NoTestImportRuleOptions {
+  testFilePattern?: string;
+}
+export const NO_TEST_IMPORT = 'NO_TEST_IMPORT';
+const DEFAULT_TEST_FILE_PATTERN = '\\.(test|spec)(\\.\\w+)?$';
+
+export default {
+  meta: {
```

Comment:
i think you can/should set the default options here?  `defaultOptions` or something?  (vs the code below)

Reply:
i could be wrong, by checking the [document](https://eslint.org/docs/latest/extend/custom-rules) it doesn't seem to be a way to set defaultOptions. i did a minor adjustment and added a schema for options validation purpose.